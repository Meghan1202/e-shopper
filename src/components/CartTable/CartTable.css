.Cart__Table__Container{
    padding-top: 0.9rem;
    overflow-x: auto;
}

.Item__Header{
    padding-left: 0.62rem;
    padding-right: 0.62rem;
}

.Cart__Table{
    margin: auto;
    border-collapse: collapse;
    border: 1px solid #ddd;
}

.Row__Header {
    background-color: gray;
    color: white;
    font-weight: bold;
    padding: 0.7rem;
}

.Item__Row {
    background-color: white;
    color: black;
    font-weight: bold;
    padding: 0.62rem;
}

.Brand__Name {
    font-size: 0.75rem;
    color: gray;
}