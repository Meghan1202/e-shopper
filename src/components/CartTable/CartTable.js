import React from 'react';
import PropTypes from 'prop-types';
import './CartTable.css';

const CartTable = ({ cartItems }) => (
  <>
    <div className="Cart__Table__Container">
      <table className="Cart__Table">
        <thead>
          <tr>
            <th className="Item__Header">Item Description</th>
            <th className="Item__Header">Unit Price</th>
            <th className="Item__Header">Quantity</th>
            <th className="Item__Header">Sub Total</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td className="Row__Header" colSpan="4">Fruits & Vegitables</td>
          </tr>
          {cartItems.map((cartItem) => (
            <tr key={cartItem.id}>
              <td className="Item__Row">
                <span className="Brand__Name">{cartItem.companyName}</span>
                <br />
                {cartItem.productName}
              </td>
              <td className="Item__Row">{cartItem.price}</td>
              <td className="Item__Row">{cartItem.count}</td>
              <td className="Item__Row">{cartItem.count * cartItem.price}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  </>
);

CartTable.propTypes = {
  cartItems: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    companyName: PropTypes.string.isRequired,
    productName: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    quantity: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
  })).isRequired,
};

export default CartTable;
