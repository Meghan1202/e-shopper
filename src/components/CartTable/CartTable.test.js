import { render } from '@testing-library/react';
import React from 'react';
import CartTable from './CartTable';

describe(CartTable.name, () => {
  const mockCartItems = {};
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('should match snapshot for the table', () => {
    const { container } = render(<CartTable cartItems={mockCartItems} />);
    expect(container).toMatchSnapshot();
  });
});
