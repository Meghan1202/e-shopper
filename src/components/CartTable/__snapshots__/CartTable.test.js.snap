// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTable should match snapshot for the table 1`] = `
<div>
  <div
    class="Cart__Table__Container"
  >
    <table
      class="Cart__Table"
    >
      <thead>
        <tr>
          <th
            class="Item__Header"
          >
            Item Description
          </th>
          <th
            class="Item__Header"
          >
            Unit Price
          </th>
          <th
            class="Item__Header"
          >
            Quantity
          </th>
          <th
            class="Item__Header"
          >
            Sub Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="Row__Header"
            colspan="4"
          >
            Fruits & Vegitables
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
